File: src/python/main_publisher.py
from publisher import Publisher
import time
import json  # Add import for JSON handling
import cv2  # Add import for OpenCV

if __name__ == "__main__":
    pub = Publisher()
    angle = 0  # Initialize rotation angle
    try:
        # Load image
        frame = cv2.imread("../../assets/pong.png")
        assert frame is not None, "❌ No se pudo cargar la imagen: pong.png"
            
        while True:
            frames = []
            for _ in range(3):  # Add 3 rotated images
                angle = (angle + 10) % 360  # Increment angle
                center = (frame.shape[1] // 2, frame.shape[0] // 2)
                matrix = cv2.getRotationMatrix2D(center, angle, 1.0)
                rotated_frame = cv2.warpAffine(frame, matrix, (frame.shape[1], frame.shape[0]))
                frames.append(rotated_frame)

            # Additional data
            data = {"key": "probando"}

            # Build and publish message with multiple images and additional data
            message_bytes = pub.build_message(frames, data)
            pub.publish_message(message_bytes)

            time.sleep(.1)
            
    except KeyboardInterrupt:
        print("\n🛑 Publicador detenido.")
        pub.close()

File: src/go/main_publisher.go
package main

import (
    "fmt"
    "image"
    "time"
    "github.com/disintegration/imaging"
    // Import the publisher package using a relative path
    "./publisher/publisher"
)

func main() {
    pub, err := publisher.NewPublisher("tcp://*:5555", "camera/image_raw")
    if err != nil {
        fmt.Println("Error creating publisher:", err)
        return
    }
    defer pub.Close()

    img, err := imaging.Open("../../assets/pong.png")
    if err != nil {
        fmt.Println("Error loading image:", err)
        return
    }

    angle := 0.0
    for {
        frames := []image.Image{
            imaging.Rotate(img, angle, image.Black),
            imaging.Rotate(img, angle+10, image.Black),
            imaging.Rotate(img, angle+20, image.Black),
        }
        data := map[string]interface{}{
            "key": "probando",
        }

        messageBytes, err := pub.BuildMessage(frames, data)
        if err != nil {
            fmt.Println("Error building message:", err)
            return
        }

        err = pub.PublishMessage(messageBytes)
        if err != nil {
            fmt.Println("Error publishing message:", err)
            return
        }

        angle += 10
        time.Sleep(100 * time.Millisecond)
    }
}
